File: src/main/java/selectionsort/SelectionSort.java
package selectionsort;

public class SelectionSort {
    public static void main(String[] args) {
        int[] inputArray = new int[]{5, 3, 6, 2, 10};
        selectionSort(inputArray);
        for (int i = 0; i < inputArray.length; i++) {
            System.out.printf("%d ", inputArray[i]);
        }
    }

    private static void selectionSort(int[] array) {
        for (int i = 0; i < array.length; i++) {
            var smallest = i;
            for (int j = i + 1; j < array.length; j++) {
                if (array[j] < array[smallest]) {
                    smallest = j;
                }
            }
            var temp = array[i];
            array[i] = array[smallest];
            array[smallest] = temp;
        }
    }
}

File: src/main/java/selectionsort/selection_sort.py
def selection_sort(arr):
    for i in range(len(arr)):
        smallest = i
        for j in range(i + 1, len(arr)):
            if arr[j] < arr[smallest]:
                smallest = j
        temp = arr[i]
        arr[i] = arr[smallest]
        arr[smallest] = temp


array = [5, 3, 6, 2, 10]
selection_sort(array)
print(array)
